widget/check.go

```diff
@@ -100,6 +100,31 @@ type Check struct {
 
 	focused bool
 	hovered bool
+
+	checkSource   binding.Bool
+	checkListener binding.DataListener
+}
+
+// Bind connects the specified data source to this Check.
+// The current value will be displayed and any changes in the data will cause the widget to update.
+// User interactions with this Check will set the value into the data source.
+//
+// Since: 2.0.0
+func (c *Check) Bind(data binding.Bool) {
+	c.Unbind()
+	c.checkSource = data
+
+	c.checkListener = binding.NewDataListener(func() {
+		c.Checked = data.Get()
+		if cache.IsRendered(c) {
+			c.Refresh()
+		}
+	})
+	data.AddListener(c.checkListener)
+
+	c.OnChanged = func(b bool) {
+		data.Set(b)
+	}
```

Comment:
```suggestion
	c.OnChanged = data.Set
```

Reply:
Thanks, updated and I realised there was an issue with constructors differing from Bind() code - simplified.